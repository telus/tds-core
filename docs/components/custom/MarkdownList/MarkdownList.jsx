import React from 'react'
import PropTypes from 'prop-types'

import Box from '../../../../packages/Box/Box'
import Text from '../../../../packages/Text/Text'
import UnorderedList from '../../../../packages/UnorderedList/UnorderedList'
import OrderedList from '../../../../packages/OrderedList/OrderedList'

// This interface must match the React Styleguidist Markdown List component
// https://github.com/styleguidist/react-styleguidist/blob/master/src/rsg-components/Markdown/List/ListRenderer.js
const MarkdownList = ({ ordered, children }) => {
  const ListComponent = ordered ? OrderedList : UnorderedList

  return (
    <Box below={3}>
      <Text block size="medium">
        <ListComponent>
          {React.Children.map(children, li => (
            <ListComponent.Item>{li.props.children}</ListComponent.Item>
          ))}
        </ListComponent>
      </Text>
    </Box>
  )
}

MarkdownList.propTypes = {
  ordered: PropTypes.bool,
  children: PropTypes.node.isRequired,
}
MarkdownList.defaultProps = {
  ordered: false,
}

export default MarkdownList
