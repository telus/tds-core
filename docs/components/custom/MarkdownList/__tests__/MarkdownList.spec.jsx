import React from 'react'
import { shallow } from 'enzyme'

import MarkdownList from '../MarkdownList'

describe('MarkdownHeading', () => {
  it('can render an unordered list', () => {
    const markdownList = shallow(
      <MarkdownList>
        <li>Item 1</li>
        <li>Item 1</li>
      </MarkdownList>
    )

    expect(markdownList).toMatchSnapshot()
  })

  it('can render an ordered list', () => {
    const markdownList = shallow(
      <MarkdownList ordered={true}>
        <li>Item 1</li>
        <li>Item 1</li>
      </MarkdownList>
    )

    expect(markdownList).toMatchSnapshot()
  })
})
