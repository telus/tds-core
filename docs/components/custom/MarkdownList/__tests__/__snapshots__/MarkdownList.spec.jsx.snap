// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownHeading can render an ordered list 1`] = `
<Box
  below={3}
  inline={false}
  tag="div"
>
  <Text
    block={true}
    bold={false}
    invert={false}
    size="medium"
  >
    <OrderedList
      listStyle="decimal"
      size="medium"
    >
      <OrderedList.Item
        key=".0"
        size="medium"
      >
        Item 1
      </OrderedList.Item>
      <OrderedList.Item
        key=".1"
        size="medium"
      >
        Item 1
      </OrderedList.Item>
    </OrderedList>
  </Text>
</Box>
`;

exports[`MarkdownHeading can render an unordered list 1`] = `
<Box
  below={3}
  inline={false}
  tag="div"
>
  <Text
    block={true}
    bold={false}
    invert={false}
    size="medium"
  >
    <UnorderedList
      listStyle="circle"
      size="medium"
    >
      <UnorderedList.Item
        key=".0"
        listStyle="circle"
        size="medium"
      >
        Item 1
      </UnorderedList.Item>
      <UnorderedList.Item
        key=".1"
        listStyle="circle"
        size="medium"
      >
        Item 1
      </UnorderedList.Item>
    </UnorderedList>
  </Text>
</Box>
`;
